components/style/properties/longhand/font.mako.rs

```diff
@@ -344,6 +344,73 @@ ${helpers.single_keyword("font-variant",
     }
 </%helpers:longhand>
 
+<%helpers:longhand products="gecko" name="font-synthesis" animatable="False">
+    use cssparser::ToCss;
+    use std::fmt;
+    use values::LocalToCss;
+    use values::computed::ComputedValueAsSpecified;
+    use values::NoViewportPercentage;
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    pub mod computed_value {
+        pub use super::SpecifiedValue as T;
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct SpecifiedValue {
+        weight: bool,
+        style: bool,
+    }
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if self.weight {
+                try!(dest.write_str("weight"));
+            }
+            else if self.style {
+                if self.weight {
```

Comment:
This branch is dead, if `self.weight` is true we'd have entered in the first `if`, not in this block.

I think the more readable way to organize this is something like:

```rust
if self.weight && self.style {
    // ...
} else if self.weight {
    // ...
} else if self.style {
    // ...
} else {
    // ...
}
```

You can of course complicate it a little more to share code, but I doubt it's useful in this case.